### ISSUE TITLE
Implement 2 minutes beacon node sync with checkpoint state

### ISSUE THREAD

**Issue by remyroy** [status-im/nim-beacon-chain#2530]
**Is your feature request related to a problem? Please describe.**
It can take a few hours to sync a Nimbus beacon node on mainnet or any of the testnet from scratch.

**Describe the solution you'd like**
It should take only 2 minutes to sync a Nimbus beacon node from scratch using a recent finalized checkpoint state.

**Additional context**
See [this video by Adrian Sutton](https://www.youtube.com/watch?v=ce9uVRl23zI) from the Teku development team. You can get more details on [the related Teku documentation page](https://docs.teku.pegasys.tech/en/latest/HowTo/Get-Started/Checkpoint-Start/).

**Comment by remyroy** [comment]
Looking at the command line flags, it seems like this might already be implemented, but I'm not sure how exactly you would use it:

```
     --weak-subjectivity-checkpoint  Weak subjectivity checkpoint in the format
                               block_root:epoch_number.
     --finalized-checkpoint-state  SSZ file specifying a recent finalized state.
     --finalized-checkpoint-block  SSZ file specifying a recent finalized block.
```

Maybe this need just a little more documentation with some examples.

**Comment by mratsim** [comment]
It's a feature that is still being worked on, hence we don't publicize it at the moment.

**Comment by arnetheduck** [comment]
In particular, there are unresolved spec issues before it can be implemented in a secure and sustainable way - see https://github.com/ethereum/eth2.0-specs/pull/2131 - teku currently operates in a grey zone of the spec when it syncs the way it does.

**Comment by arnetheduck** [comment]
Another way to start up quickly is to  simply copy the existing sqlite database - it's one file that's fairly easy to move around.

**Comment by arnetheduck** [comment]
https://notes.status.im/nimbus-eth2-checkpoint-sync# outlines a rough technical roadmap for this feature

**Comment by arnetheduck** [comment]
#3209

**Comment by arnetheduck** [comment]
Released in 1.7.0: https://nimbus.guide/trusted-node-sync.html
